### DISCUSSION TITLE
Don't use port `8080` for HTTPS; breaks conventions

### DISCUSSION THREAD

**Discussion by francislavoie** [nextcloud/all-in-one#4003] [Questions] [answered]
### Steps to reproduce
1. Get asked for help from users of Caddy on https://caddy.community/
2. See that all-in-one serves port 8080
3. Okay cool, let's proxy to that port, that's easy, it should be serving HTTP
4. See "You're speaking plain HTTP to an SSL-enabled server port."
5. ??????
6. Port `8080` has `80` in it, which means "HTTP". It should _NOT_ be serving HTTPS, that breaks conventions.
7. A different port like 8443 should be used conventionally for HTTPS traffic.

### Expected behavior 

Port 8080 should serve unencrypted HTTP, not self-signed HTTPS.

### Actual behavior 

Port 8080 serves HTTPS. It should not.

It seems like some users are using a workaround by having Apache listen on port `11000` for HTTP. That's weird and confusing. Why not just serve HTTP by default to make it easy for the reverse proxy usecase? Why recommend an unconventional port like 11000 for HTTP?

**Comment by Zoey2936** [comment]
You need to proxy to port 11000 if you follow the reverse-proxy.md

**Comment by francislavoie** [comment]
To be clear, I'm not asking for help. I don't run Nextcloud myself. I'm just posting this out of frustration as a maintainer of Caddy getting asked for help from users who are confused about the awkward port setup that all-in-one recommends.

It's very misleading and unconventional to use HTTPS on port 8080, because `80` implies HTTP (unencrypted).

The fact that this was chosen means that a different port needs to be used for HTTP (apparently 11000) which is even weirder because that is not a conventional port.

**Comment by Zoey2936** [comment]
80 - http redirect to 443 https
443 - signed https nextcloud port, unused in reverse proxy mode
8443 - aio https port
8080 - aio self signed https port
11000 - nextcloud http port, used in reverse proxy mode, the proxy needs to proxy to this port

Changing the default ports currently used (8080, 8443, 11000) would maybe break existing instances

**Comment by francislavoie** [comment]
> 8080 - aio self signed https port

Yeah, that's very dumb.

I understand it's too late to change at this point, but my goodness, what a terrible decision that was.

**Comment by Zoey2936** [comment]
But still they need to proxy to 11000 http not 8080

**Comment by francislavoie** [comment]
But users would be much less confused if all-in-one used conventions (HTTP on 8080, not HTTPS) which would be predictable and easy to understand.

**Comment by Zoey2936** [comment]
even if 8080 would be http, they still need to proxy to 11000

**Comment by francislavoie** [comment]
Why would that be the case? I don't find this easy to understand at all.

**Comment by Zoey2936** [comment, marked as answer]
AIO has two parts:
AIO and nextcloud
The aio container is used to manage the nextcloud container and its services

aio uses 8080, 8443 for the management UI and if exposed 80 for redirects

nextcloud 443 in default mode without reverse proxy support and a custom http port in reverse proxy mode (11000 used in the documentation)

aio don't need to be proxied since it is an admin UI and nextcloud needs to be proxied in reverse proxy mode

**Comment by francislavoie** [comment]
That's very confusing. I didn't find that the docs made that clear at all.

Also proxying to `8080` serves a page like this, which is not easy to understand either:

![image](https://github.com/nextcloud/all-in-one/assets/2111701/cae81777-3534-4770-8fd5-7f3c1328cad7)

**Comment by szaimen** [comment]
Hi @francislavoie I am glad that you found your way here 👋

Indeed AIO works very different compared to other docker projects. The main idea is to make things easier for people unfamiliar with docker by doing so.

You can look how things actually work under the hood by looking at the [manual install](https://github.com/nextcloud/all-in-one/tree/main/manual-install): https://github.com/nextcloud/all-in-one/blob/main/manual-install/latest.yml

We would gladly improve our documentation if you can give us specific pointers what and how to improve things in detail.

Thanks a lot and have a nice day!
